compiler/rustc_middle/src/query/mod.rs

```diff
@@ -1039,7 +1043,18 @@ rustc_queries! {
     /// then `Err(AlwaysRequiresDrop)` is returned.
     query adt_drop_tys(def_id: DefId) -> Result<&'tcx ty::List<Ty<'tcx>>, AlwaysRequiresDrop> {
         desc { |tcx| "computing when `{}` needs drop", tcx.def_path_str(def_id) }
-        cache_on_disk_if { true }
+        cache_on_disk_if { false }
+    }
+
+    /// A list of types where the ADT requires drop if and only if any of those types
+    /// has significant drop. A type marked with the attribute `rustc_insignificant_dtor`
+    /// is will always considered to not be significant. A drop is significant if it is
```

Comment:
personally, I'd remove the word always. It kind of gives me an illusion that the query will on its own, try to check some criterion to decide if the drop is significant and the attribute is some form of early exit/helper annotation for corner cases.